osu.Game/Screens/OnlinePlay/Multiplayer/Match/MultiplayerReadyButton.cs

```diff
@@ -187,25 +232,219 @@ private void updateState()
             });
         }
 
-        private void updateButtonColour(bool green)
+        public class ReadyButton : Components.ReadyButton
         {
-            if (green)
+            public new Triangles Triangles => base.Triangles;
+
+            [Resolved]
+            private MultiplayerClient multiplayerClient { get; set; }
+
+            [Resolved]
+            private OsuColour colours { get; set; }
+
+            [CanBeNull]
+            private MultiplayerRoom room => multiplayerClient.Room;
+
+            protected override void LoadComplete()
             {
-                button.BackgroundColour = colours.Green;
-                button.Triangles.ColourDark = colours.Green;
-                button.Triangles.ColourLight = colours.GreenLight;
+                base.LoadComplete();
+
+                multiplayerClient.RoomUpdated += () => Scheduler.AddOnce(onRoomUpdated);
+                onRoomUpdated();
             }
-            else
+
+            protected override void Update()
+            {
+                base.Update();
+
+                if (room?.Countdown != null)
+                {
+                    // Update the countdown timer.
+                    onRoomUpdated();
```

Comment:
I'd much prefer this was using `Scheduler.AddDelayed` logic to limit updates to less often than every update call. `updateButtonText` is doing string construction and LINQ, so it'll probably add up.